test/jdk/javax/print/UnixPrintJobOptionsTest.java

```diff
@@ -0,0 +1,157 @@
+            return;
+        }
+
+        PrintStream origPrintStream = System.out;
+
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        PrintStream printStream = new PrintStream(os);
+        System.setOut(printStream);
+
+        boolean success = false;
+
+        for (PrintService testPrintService : printerSupportedAttrs.keySet()) {
+            String patternStr = "/usr/bin/lpr\\s+.*";
+            PrintRequestAttributeSet attributeSet = new HashPrintRequestAttributeSet();
+            for (Class<? extends PrintRequestAttribute> attrClass : printerSupportedAttrs.get(testPrintService)) {
+                Object supportedValues = testPrintService.getSupportedAttributeValues(attrClass, docFlavor, attributeSet);
+                if (supportedValues == null || ((Object[]) supportedValues).length == 0) {
+                    continue;
+                }
+                if (attrClass == NumberUp.class) {
+                    NumberUp numberUp = ((NumberUp[]) supportedValues)[0];
+                    patternStr = patternStr + "\\s-o number-up=" + numberUp.getValue();
+                    attributeSet.add(numberUp);
+                } else if (attrClass == OrientationRequested.class) {
+                    for (OrientationRequested orientationRequested : ((OrientationRequested[]) supportedValues)) {
+                        if (orientationRequested != OrientationRequested.PORTRAIT) {
+                            patternStr = patternStr + "\\s-o orientation-requested=" + orientationRequested.getValue();
+                            attributeSet.add(orientationRequested);
+                            break;
+                        }
+                    }
+                } else if (attrClass == Sides.class) {
+                    Sides s = ((Sides[]) supportedValues)[0];
+                    patternStr = patternStr + "\\s-o sides=" + s.toString();
+                    attributeSet.add(s);
+                }
+            }
+
+            if (attributeSet.size() < 2) {
+                continue;
+            }
+            System.out.println("Debug pattern: " + patternStr);
+            DocPrintJob docPrintJob = testPrintService.createPrintJob();
+            try {
+                docPrintJob.print(new SimpleDoc("UnixPrintJob options test",
+                        docFlavor, new HashDocAttributeSet()), attributeSet);
+            } catch (PrintException ex) {
+                throw new RuntimeException(ex);
+            } finally {
+                System.setOut(origPrintStream);
+
+            }
+            String debug = os.toString();
+            System.out.print(debug);
+            Pattern pattern = Pattern.compile(patternStr);
+            if (!pattern.matcher(debug).find()) {
+                throw new RuntimeException("Output does not contain necessary options");
+            } else {
+                success = true;
+                break;
+            }
+        }
+        if (!success) {
```

Comment:
Maybe it would be more robust to wrap everything in a try block with a finally to restore the printstream.